docs/python.rst

```diff
@@ -0,0 +1,78 @@
+Python domain customization
+===========================
+
+.. confval:: python_type_aliases
+
+   Maps names or dotted names appearing in type annotations in Python signatures
+   to the corresponding fully-qualified name.
+
+   For example, to map :python:`MyUnqualifiedType` to :python:`alias_ex.MyUnqualifiedType`, add the following
+   to :file:`conf.py`:
+
+   .. literalinclude:: conf.py
+      :language: python
+      :start-after: # BEGIN: python_type_aliases example
+      :end-before: # END: python_type_aliases example
+
+   .. rst-example:: Python signature modified by type alias
+
+      .. py:function:: foo(x: MyUnqualifiedType) -> str
+         :noindex:
+
+         Function description.
+
+.. confval:: python_resolve_unqualified_typing
+
+   Specifies whether to add the following mappings to
+   :confval:`python_type_aliases`:
+
+   .. jinja:: sys
+
+      {%- for x in sys.modules["sphinx_immaterial.python_type_annotation_transforms"].TYPING_NAMES %}
+      - :python:`{{ x }}` -> :py:obj:`typing.{{ x }}`
+      {%- endfor %}
+
+   If :confval:`python_transform_type_annotations_pep585` is also set to
+   :python:`True`, then these aliases are additionally subject to that mapping.
+
+.. confval:: python_transform_type_annotations_pep585
+
+   Specifies whether to add the following :pep:`585` mappings to
+   :confval:`python_type_aliases`:
+
+   .. jinja:: sys
+
+      {%- for k, v in sys.modules["sphinx_immaterial.python_type_annotation_transforms"].PEP585_ALIASES.items() %}
+      - :py:obj:`{{ k }}` -> :py:obj:`{{ v }}`
+      {% endfor %}
```

Comment:
![image](https://user-images.githubusercontent.com/14963867/165832922-8b74668b-8a9b-4bd8-8486-b1dda64a006c.png)
Its not too important to me because users can type hint directly from the `re` namespace. However, I'm see unresolved references here

Reply:
Yeah, I think that is just because the Python docs don't include those symbols.

Reply:
Best to keep it as is then to demonstrate py-docs' shortcomings then.